avm/res/container-service/managed-cluster/README.md

```diff
@@ -3292,6 +3335,21 @@ A CIDR notation IP range from which to assign service cluster IPs. It must not o
 - Required: No
 - Type: string
 
+### Parameter: `skuName`
+
+Name of a managed cluster SKU.
+
+- Required: No
+- Type: string
+- Default: `'Base'`
+- Allowed:
+  ```Bicep
+  [
+    'Automatic'
+    'Base'
+  ]
+  ```
+
```

Comment:
There are more requirements to get an AKS automatic cluster up and running? Have you verified the cluster creation? During my initial prototype additional configuration was required.

Please review my original PR for `azd`: https://github.com/Azure/azure-dev/pull/4069/files#diff-b13168826911b7f1ec80445aa91011c9a0e3430e1f4555cdb9276a141d877872R23

Reply:
Perhaps a new `automatic-cluster` pattern module would be a good fit to preset all the automatic cluster configuration requirements.

Reply:
@zedy-wj 
Pattern modules are usually collections of modules. Do you think that is the case here?, what modules would be required in that pattern module to support automatic?

Reply:
> There are more requirements to get an AKS automatic cluster up and running? Have you verified the cluster creation? During my initial prototype additional configuration was required.
> 
> Please review my original PR for `azd`: https://github.com/Azure/azure-dev/pull/4069/files#diff-b13168826911b7f1ec80445aa91011c9a0e3430e1f4555cdb9276a141d877872R23

Hey @wbreza,
Sorry for the delay. Regarding the AKS automatic cluster that you mentioned requires some additional configuration, we have added it in the previous PR: https://github.com/Azure/bicep-registry-modules/pull/2787 based on your original PR. In that PR, we missed the configuration:
```
  sku: {
    name: 'Automatic'
  }
```
 so this PR is actually a supplement to previous PR. Of course, we have written tests to test the creation of the cluster, which you can view here: https://github.com/Azure/bicep-registry-modules/pull/3104/files#diff-05e25c8270f0f2c149ecaa0542411e6191dfd6fe4d57cc2f04c5a469f315f7c7R35-R90. In the pipeline, it can pass normally.

Reply:
> @zedy-wj Pattern modules are usually collections of modules. Do you think that is the case here?, what modules would be required in that pattern module to support automatic?

Hey @jongio,
In the `aks-automatic-cluster.bicep` file added in the original [PR](https://github.com/Azure/azure-dev/pull/4069/files#diff-b13168826911b7f1ec80445aa91011c9a0e3430e1f4555cdb9276a141d877872), I only found the creation of `automatic-cluster`, so I don't think it is more suitable to be changed to pattern mudule. 

Besides, the current actual practice in original PR is: a new `boolean` type parameter `automatic` is introduced in the `aks.bicep` file, and `automatic-cluster` is created only when it is set to `true`. In the `Pattern modules` we have created so far, we have converted the `aks.bicep` file in `infra/core` into a `Pattern module`: `azd/aks`. Please refer to this PR: https://github.com/Azure/bicep-registry-modules/pull/3250 for details. Therefore, I think the current best practice is: in the original `container-service\managed-cluster` res module Introduce a new skuName `automatic` (this PR currently does this, and the relevant configuration of automatic has been merged in PR: https://github.com/Azure/bicep-registry-modules/pull/2787), and add the `automatic` parameter in the `Pattern module`: `azd/aks` to determine which kind of clusters to create (we will update the `Pattern module`: `azd/aks` after this PR merge). In this way, we can restore all the content in `infra/core` and [original PR](https://github.com/Azure/azure-dev/pull/4069) to the greatest extent without adding additional content.

Reply:
@wbreza - Can you review and come up with a recommendation for this?